app/services/users_service.rb

```diff
@@ -41,6 +34,24 @@ def register(email, password, organization_name)
     result
   end
 
+  def register_from_invite(email, password, organization_id)
+    result.user = User.find_or_initialize_by(email: email)
+
+    if result.user.id
+      result.fail!(code: 'user_already_exists')
+
+      return result
+    end
+
+    ActiveRecord::Base.transaction do
```

Comment:
What about moving the transaction inside `create_user_and_membership`method?

Reply:
Done!